### ISSUE TITLE
Use requestAnimationFrame for applying the transition

### ISSUE THREAD

**Issue by roblevintennis** [JSteunou/marionette.csstransition-region#1]
I was looking at this issue in our codebase and noticed you had it too in these places:
https://github.com/JSteunou/marionette.csstransition-region/blob/master/marionette.csstransition-region.js#L59
https://github.com/JSteunou/marionette.csstransition-region/blob/master/marionette.csstransition-region.js#L68

I iterated first using `_.defer` (you can't use since you're not trying to include underscore dependency of course), but, which is ultimately setTimeout with zero ms. As explained to me, it essentially waits for whatever's on the call stack at that moment to complete. Anyway, it worked. Then I thought to try request animation frame which is the best practice these days. That also worked for me in chrome and firefox.

Fwiw, in our coffeescript based codebase, we shimmed it with:

```
window.requestAnimationFrame = do ->
  window.requestAnimationFrame ||
  window.webkitRequestAnimationFrame ||
  window.mozRequestAnimationFrame ||
  window.oRequestAnimationFrame ||
  window.msRequestAnimationFrame
```

which is essentially tranposing of Paul Irish's http://www.paulirish.com/2011/requestanimationframe-for-smart-animating/

Hope this helps man :)

**Comment by JSteunou** [comment]
Yep, figure out I should have used raf months ago but that project was just a PoC, never updated since, nor with last marionette neither with all fixes I could think about. But thank you for pointed this out, I will let this open, it could help some people still reading this PoC.

**Comment by roblevintennis** [comment]
Sounds good!
